### ISSUE TITLE
routinator no longer listening

### ISSUE THREAD

**Issue by ccakes** [NLnetLabs/routinator#106]
I'm running routinator built from master (older commit though - dc2a4be0af2364a1f8364e167ed7dcd91f23025c).

We reloaded a router this morning and the RTR session didn't re-establish, when I started digging in I noticed the server was no longer listening on the configured TCP port. For reference, we're listening on 8323. Some output from the server:

```
$ netstat -tanp | grep ':8323' | grep LISTEN
$ netstat -tanp | grep ':8323' | grep ESTABLISHED | wc -l
50
$ cat /var/lib/routinator/rtrd.pid
5709
$ ps aux | grep 5709
routina+  5709  0.0  0.6  68452 50240 ?        Sl   Mar21  20:45 /usr/local/bin/routinator -c /etc/routinator/config.toml rtrd
root     22632  0.0  0.0  11112   940 pts/2    R+   09:24   0:00 grep 5709
```

From this output you can see that nothing is listening on 8323 however there are still a number of established sessions.. the process was still running and there was nothing useful in the logs. After restarting the daemon, all sessions came back up and it's now listening happily on 8323

```
$ netstat -tanp | grep ':8323' | grep LISTEN
tcp        0      0 0.0.0.0:8323            0.0.0.0:*               LISTEN      33196/routinator 
```

I've increased the logging verbosity to debug and will see if it happens again, but is this something you've seen before? I couldn't find anything obvious searching the issues here

**Comment by AlexanderBand** [comment]
My Routinator died too since this morning:
```
Found valid trust anchor rsync://rpki.apnic.net/repository/apnic-rpki-root-iana-origin.cer. Processing.
Found valid trust anchor rsync://rpki.arin.net/repository/arin-rpki-ta.cer. Processing.
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:345:21
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
Found valid trust anchor rsync://repository.lacnic.net/rpki/lacnic/rta-lacnic-rpki.cer. Processing.
```
The process still runs, but the RTR and HTTP servers are no longer listening.

**Comment by partim** [comment]
I have at least two more instances. Looks like this happened to all routinator instances at around 2019-04-01T01:19:55Z. Some seem to have died outright, some only lost their RTR (and HTTP?) service.

**Comment by partim** [comment]
Also see #107

**Comment by partim** [comment]
This got fixed in [rpki-rs’ #30](https://github.com/NLnetLabs/rpki-rs/pull/30). A release will follow soon.
